src/nvim/ex_getln.c

```diff
@@ -2392,6 +2418,134 @@ static void cmdpreview_close_win(void)
   }
 }
 
+/// Save current state and prepare windows and buffers for command preview.
+static void cmdpreview_prepare(CpInfo *cpinfo)
+{
+  kv_init(cpinfo->buf_info);
+  kv_init(cpinfo->win_info);
+
+  FOR_ALL_BUFFERS(buf) {
```

Comment:
Do we need to do this, or is it enough to save state for each buffer displayed in a window in the current tab? Or is the idea that commands want to operate even on hidden buffers?

Reply:
> Do we need to do this, or is it enough to save state for each buffer displayed in a window in the current tab? Or is the idea that commands want to operate even on hidden buffers?

It is enough to save state for each buffer displayed in a window, but it's more complicated than just doing it for each buffer since the order is important here.

Reply:
> It is enough to save state for each buffer displayed in a window, but it's more complicated than just doing it for each buffer since the order is important here.

This is fine, but would be good to mention in a comment in the code right here :)

Reply:
I tried saving buffer and window state in the same loop to check if it works, and all the tests pass locally for me, so let's just go with that for now since that'd be the simplest solution.